src/Facade/Services/StockQuantitiesService.cs

```diff
@@ -1,35 +1,23 @@
 ﻿namespace Linn.Stores.Facade.Services
 {
     using System.Collections.Generic;
-    using System.Text.RegularExpressions;
 
     using Linn.Common.Facade;
-    using Linn.Common.Persistence;
+    using Linn.Stores.Domain.LinnApps.ExternalServices;
     using Linn.Stores.Domain.LinnApps.StockLocators;
 
     public class StockQuantitiesService : IStockQuantitiesService
     {
-        private readonly IQueryRepository<StockQuantities> repository;
+        private readonly IFilterByWildcardQueryRepository<StockQuantities> repository;
 
-        public StockQuantitiesService(IQueryRepository<StockQuantities> repository)
+        public StockQuantitiesService(IFilterByWildcardQueryRepository<StockQuantities> repository)
         {
             this.repository = repository;
         }
 
         public IResult<IEnumerable<StockQuantities>> GetStockQuantities(string partNumber)
         {
-            if (!partNumber.Contains("*"))
-            {
-                var res = this.repository
-                    .FilterBy(x => x.PartNumber == partNumber.ToUpper());
-                return new SuccessResult<IEnumerable<StockQuantities>>(res);
-            }
-
-            var partNumberPattern = Regex.Escape(partNumber.Trim(' ')).Replace("\\*", ".*?");
-            var r = new Regex(partNumberPattern, RegexOptions.IgnoreCase);
-
-            var result = this.repository
-                .FilterBy(x => r.IsMatch(x.PartNumber));
+            var result = this.repository.FilterByWildcard(partNumber.Replace("*", "%").ToUpper());
```

Comment:
So does the EF.Functions.Like use "%" like oracle as the wildcard character? If so then that's pretty nice, and this'll let users either use % if they're used to that or * if they prefer it